### DISCUSSION TITLE
WORKAROUND: Abort error while syncing: Filename contains illegal characters (Mac to PC)

### DISCUSSION THREAD

**Discussion by SAI-GIT1** [remotely-save/remotely-save#176] [Show and tell]
**_remotely-save-1660771970923: abort sync, triggerSource=manual, error while syncing_**

Using Obsidian with remotely-save and Dropbox on a Mac, I created new notes that contained illegal characters for PC. They were: question mark, ? and double-quotes, ""

Syncing worked fine on the Mac, but when I went to the PC, remotely-save produced the above abort error. Inspecting Console with Ctrl-Shift-I revealed the names of the two offending files.

Even though those files had been deleted on the Mac, they continued to exist, most likely due to the Delete File handling option in Obsidian to move the files in an Obsidian .trash folder.

The successful workaround was to go back to the Mac, recreate the text files with the offending names, change the Obsidian file and link settings to Permanently delete files, sync with remotely-sync, delete the offending files, sync again with remotely-sync.
Upon returning to the PC and syncing, SUCCESS.

Seems like a chunk of code needs to be added to either convert offending filenames to escape characters internally or prevent them from being used on all platforms. Those both could be potential programmatic fixes.

Hope this helps!

**Comment by fyears** [comment]
Thanks for your detailed report!

And as you said, unfortunately this plugin doesn't consider special characters cross platforms carefully, it just upload and download them as is...
